internal/init/prompts.go

```diff
@@ -88,6 +88,36 @@ func ValidateSAK(input string) error {
 	return nil
 }
 
+// ValidatePname validates Project Name field string input.
+func ValidatePname(input string) error {
+	// the first 62 char out of base64 and -
+	var pName = regexp.MustCompile(`^[A-Za-z0-9-]{1,15}$`)
```

Comment:
any thoughts on capping this at 15 chars. 

we briefly discussed possibly generating a shortened slug that `tf` can use instead of the project-name till that's done what do you think about capping this?

Reply:
Does 15 chars work? no conflicts? 

If we were to limit the char length I would actually separate the return in 2 parts 
an if statement to check the len 
```go 
// perhaps something like this
if len(str) > maxLength {
  return errors.New("Exceeded maximum length blahblah")
}
if !pName.MatchString(input) { // other rules ...
```

Reply:
I do think 15 chars will be too limiting, but it should be fine for now.

Reply:
This is actually 4 chars too long see 👇

```shell
Error: expected length of name_prefix to be in the range (1 - 32), got dtoki-infra-015-production-us-west-2

  on .terraform/modules/production.eks.eks/terraform-aws-eks-12.1.0/cluster.tf line 112, in resource "aws_iam_role" "cluster":
 112: resource "aws_iam_role" "cluster" {
```

There is a 32 char cap on then aws I am role if we were to move forward, the max should be 11 chars (using the prod naming schema as the ceiling ) which really sucks, some options I can think of are:

1. change our infrastructure naming scheme for production  & staging.
 eg. -prod-us-west-2; -stage-us-west-2.
 this gives us 16 chars to work with.

2. create a unique shortened slug name that `tf` uses.

what do you folks suggest?

Reply:
Yeah let's use prod and stage. Also even having the region in there is a bit overkill. Few startups will actually have multi-region setups which would require that to be part of the names.  Can you start a list in a google doc or somewhere else to track these places where we hit character limits in AWS?

Reply:
[aws limits doc](https://docs.google.com/document/d/1tNyWVx79H7N-R5JLq_7VXKceviQEFNqVoKf_hgHStHI/edit?usp=sharing)